java/core/src/test/java/co/worklytics/psoxy/rules/JavaRules2TestBaseCase.java

```diff
@@ -0,0 +1,34 @@
+package co.worklytics.psoxy.rules;
+
+import co.worklytics.psoxy.rules.google.PrebuiltSanitizerRules;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+/**
+ * base test case for rules written in java (although also expected to have yaml-encoded equivalent
+ * in repo under `src/main/resources/rules/`)
+ */
+public abstract class JavaRules2TestBaseCase extends Rules2BaseTestCase {
+
+
+    //TODO: we could avoid this if java rules serialized to YAML as part of build process or
+    // something, rather than under version control themselves; or flip so that master copy is the
+    // yaml and java ones read from file system??
```

Comment:
As a future thing, to me the java ones allows use to do probably more complex transformations, negative selections, subsets, etc. so verbosity reduced and rules easier to understand enriching the jsonPaths and transforming then to yaml. So maybe a tool that generates them... just throwing the idea.

Reply:
yeah, i think there's a future in which we do something like that.  Have a web ux in Worklytics that generates rule sets based on policies (eg, "include employee ids in plain") to allow  high level customization without granularly customizing the rule sets.